### PR TITLE
Allow cycling through severities of error list

### PR DESCRIPTION
**Pull request by timfel**
My elisp isn't very good, but this does what I want - feel free to just take it as a "functional spec" if you dislike the code.

### CODE REVIEW

File: lsp-treemacs.el [hunk 1]
```diff
@@ -273,9 +284,20 @@
 (defvar lsp-treemacs-error-list-mode-map
   (let ((m (make-sparse-keymap)))
     (define-key m (kbd "x") #'lsp-treemacs-quick-fix)
+    (define-key m (kbd "=") (lambda ()
```
Comment (yyoncho) [inline lsp-treemacs.el]: can you make this named function so the user could rebind it to something else?

File: lsp-treemacs.el [hunk 2]
```diff
@@ -273,9 +284,20 @@
...
+                              (interactive)
+                              (setq lsp-treemacs-error-list-severity
+                                    (if (= lsp-treemacs-error-list-severity 1)
+                                        3
+                                      (- lsp-treemacs-error-list-severity 1)))
```
Comment (yyoncho) [inline lsp-treemacs.el]: nit: use (1- lsp-treemacs-error-list-severity)

File: lsp-treemacs.el [hunk 3]
```diff
@@ -273,9 +284,20 @@
...
+                              (lsp-treemacs--after-diagnostics)))
     m)
   "Keymap for `lsp-treemacs-error-list-mode'.")
 
+(defvar lsp-treemacs-error-list-severity
```
Comment (yyoncho) [inline lsp-treemacs.el]: can you make this defcustom and put it at the top of the file?

**Comment by timfel** [comment]
I made the changes you requested :)

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**